workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -70,37 +46,25 @@ protected override void OnDestroyManager()
             base.OnDestroyManager();
         }
 
-        // todo replace this with something that polls for entities added and removed
         protected override unsafe void OnUpdate()
         {
-            for (var i = 0; i < addedEntitiesData.Length; i++)
+            foreach (var entityId in entitySystem.GetEntitiesAdded())
             {
-                var entity = addedEntitiesData.Entities[i];
-                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i].EntityId;
+                workerSystem.TryGetEntity(entityId, out var entity);
```

Comment:
Is this intended to discard the result of the `TryGet`? 

If so is it worth exposing a `GetEntity` method which doesn't have the out param and will actually throw if the entity doesn't exist?

Reply:
Such that the use cases of the methods break down into: 

- `GetEntity`: Entity will be there and its an error if its not.
- `TryGetEntity`: Entity may be there and I can handle the case where its not.

Reply:
Yes and maybe. The entity definitely exists at this point, however mid term I'd rather look at removing the dependency on ecs entities. GetEntity should be a method though.

Reply:
Basically I didn't touch that stuff too much as I suspect we may want to change it a lot.